packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/general.tsx

```diff
@@ -310,13 +311,42 @@ const NewDocDefaultModeSettings = () => {
 
 export const General = () => {
   const t = useI18n();
+  const { openConfirmModal } = useConfirmModal();
+  const { editorSettingService } = useServices({ EditorSettingService });
+
+  const settings = useLiveData(editorSettingService.editorSetting.settings$);
+
+  const onAIChange = useCallback(
+    (checked: boolean) => {
+      editorSettingService.editorSetting.set('enableAI', checked);
+      // refresh the page to apply the change
+      window.location.reload();
+    },
+    [editorSettingService.editorSetting]
+  );
+  const onToggleAI = useCallback(
+    (checked: boolean) => {
+      openConfirmModal({
+        title: checked ? 'Enable AI?' : 'Disable AI?',
+        description: `Are you sure you want to ${checked ? 'enable' : 'disable'} AI? This will refresh the page to apply the change.`,
+        confirmText: checked ? 'Enable' : 'Disable',
+        cancelText: 'Cancel',
+        onConfirm: () => onAIChange(checked),
+        confirmButtonOptions: {
+          variant: checked ? 'primary' : 'error',
+        },
+      });
+    },
+    [openConfirmModal, onAIChange]
+  );
+
```

Comment:
DO NOT BRING CHILDREN STATE TO PARENT ELEMENT

Reply:
I have extracted these codes to another component. 

The following codes will refactored in another PR, where the `<SettingRow>` and `i18n` will be put in specific component. cc @JimmFly 
```ts
      <SettingRow
        name={t['com.affine.settings.editorSettings.general.ai.title']()}
        desc={t['com.affine.settings.editorSettings.general.ai.description']()}
      >
        <AISettings />
      </SettingRow>
      <SettingRow
        name={t['com.affine.appearanceSettings.font.title']()}
        desc={t['com.affine.appearanceSettings.font.description']()}
      >
        <FontFamilySettings />
      </SettingRow>
      <CustomFontFamilySettings />
      <SettingRow
        name={t[
          'com.affine.settings.editorSettings.general.default-new-doc.title'
        ]()}
        desc={t[
          'com.affine.settings.editorSettings.general.default-new-doc.description'
        ]()}
      >
        <NewDocDefaultModeSettings />
      </SettingRow>
```